lib/src/dart2_suggestors/pubspec_over_react_upgrader.dart

```diff
@@ -71,25 +71,32 @@ class PubspecOverReactUpgrader implements Suggestor {
     if (overReactMatch != null) {
       // over_react is already in pubspec.yaml
       final constraintValue = overReactMatch.group(2);
-      final constraint = VersionConstraint.parse(constraintValue);
-
-      if (shouldUpdateVersionRange(
-          targetConstraint: targetConstraint,
-          constraint: constraint,
-          shouldIgnoreMin: shouldIgnoreMin)) {
-        // Wrap the new constraint in quotes if required.
-        var newValue =
-            generateNewVersionRange(constraint, targetConstraint).toString();
-
-        if (mightNeedYamlEscaping(newValue)) {
-          newValue = '"$newValue"';
+      try {
+        final constraint = VersionConstraint.parse(constraintValue);
+
+        if (shouldUpdateVersionRange(
+            targetConstraint: targetConstraint,
+            constraint: constraint,
+            shouldIgnoreMin: shouldIgnoreMin)) {
+          // Wrap the new constraint in quotes if required.
+          var newValue =
+              generateNewVersionRange(constraint, targetConstraint).toString();
+
+          if (mightNeedYamlEscaping(newValue)) {
+            newValue = '"$newValue"';
+          }
+
+          // Update the version constraint to ensure a safe minimum bound.
+          yield Patch(
+              sourceFile,
+              sourceFile.span(overReactMatch.start, overReactMatch.end),
+              '  over_react: $newValue');
         }
-
-        // Update the version constraint to ensure a safe minimum bound.
-        yield Patch(
-            sourceFile,
-            sourceFile.span(overReactMatch.start, overReactMatch.end),
-            '  over_react: $newValue');
+      } catch (e) {
+        // We can skip these they are versions we dont want to mess with in this codemod
```

Comment:
#nit Could we add the why behind this? If not, maybe just clean it up a little:
```suggestion
        // We can skip these. They are versions we don't want to mess with in this codemod.
```